File: php/Str.php
<?php
declare(strict_types=1);

namespace TinyUtils;

final class Str {
    public static function slugify(string $text): string {
        $s = mb_strtolower(trim($text));
        $s = preg_replace('/[^\w\s-]/u', '', $s);
        $s = preg_replace('/[-\s]+/u', '-', $s);
        return trim($s, '-');
    }

    public static function truncate(string $text, int $maxLen = 80, string $suffix = '...'): string {
        if (mb_strlen($text) <= $maxLen) return $text;
        $truncated = mb_substr($text, 0, $maxLen);
        $lastSpace = mb_strrpos($truncated, ' ');
        return ($lastSpace !== false ? mb_substr($truncated, 0, $lastSpace) : $truncated) . $suffix;
    }
}

File: lua/str.lua
local str = {}

function str.slugify(text)
  local s = text:lower():match("^%s*(.-)%s*$")
  s = s:gsub("[^%w%s-]", "")
  s = s:gsub("[%s-]+", "-")
  return s:match("^-+(.-)") or s
end

function str.truncate(text, max_len, suffix)
  max_len = max_len or 80
  suffix = suffix or "..."
  if #text <= max_len then return text end
  local truncated = text:sub(1, max_len)
  local last_space = truncated:match("^.*()%s")
  if last_space then return truncated:sub(1, last_space - 1) .. suffix end
  return truncated .. suffix
end

return str
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
function str.is_palindrome(text)
  local cleaned = text:lower():gsub("[^%w]", "")
  return cleaned == cleaned:reverse()
end
